scripts/generate_tutorial_readmes.py

```diff
@@ -165,14 +173,32 @@ def write_badge_table(notebooks):
         "| - | --- | ---- |",
     ]
 
-    # Add each row of the table
-    for i, local_path in enumerate(notebooks, 1):
+    # Add intro
+    intro_file = [name for name in notebooks if 'Intro' in name]
+    if len(intro_file) == 1:
+        colab_badge = make_colab_badge(intro_file[0])
+        nbviewer_badge = make_nbviewer_badge(intro_file[0])
+        table_text.append(
+            f"| Intro | {colab_badge} | {nbviewer_badge} |"
+        )
 
+    # Add tutorials
+    for i, local_path in enumerate([name for name in notebooks if 'Tutorial' in name], 1):
         colab_badge = make_colab_badge(local_path)
         nbviewer_badge = make_nbviewer_badge(local_path)
         table_text.append(
             f"| Tutorial {i} | {colab_badge} | {nbviewer_badge} |"
         )
+
+    # Add outro
+    outro_file = [name for name in notebooks if 'Outro' in name]
+    if len(outro_file) == 1:
+        colab_badge = make_colab_badge(outro_file[0])
+        nbviewer_badge = make_nbviewer_badge(outro_file[0])
+        table_text.append(
+            f"| Outro | {colab_badge} | {nbviewer_badge} |"
+        )
+
```

Comment:
Is it necessary to copy-paste this code block three times? Can't you build one master list of file names with the conditional logic for intro/tutorial/outro and then iterate over that?

Reply:
ok will try this out now